### DISCUSSION TITLE
Separate tags for SRC and binary RPM

### DISCUSSION THREAD

**Discussion by StefanBruens** [rpm-software-management/rpm#2228] [General]
Currently, several tags are used for both the src as the main binary/noarch package. Only subpackages may have separate tags.

While for some tags (e.g. description or summary) this may be just a slight inaccuracy, for other tags this may be actually problematic:

License: The upstream source tarball may bundle third party parts which fall under a different license. This may be tools just used at build time, or parts which are added just as a convenience and often get unbundled by Linux 
distributions.  
All workarounds have some tradeoffs, e.g. modifying the tarball makes  verification vs upstream sources harder. Keeping the upstream tarball makes either the binary package license to restrictive, or the src package incorrect.

BuildArch: When specifying noarch, any "%{arch}" dependent statements no longer work as some may expect. While these should not be necessary in a perfect world, sometimes building and or testing may require some arch  specific workarounds.

Of course the binary/noarch package and the src package can use different names (i.e. "main" package is empty), but this is not very convenient, typically would require renaming the src package, and is strongly discouraged by most distribution packaging 
policies.

Unfortunately, RPM disallows to create a subpackage (%package) with the same as the src package, which would solve most (all?) of these problems.

**Comment by pmatilai** [comment]
For license, we recently added separate SourceLicense tag. BuildArch is far trickier as it send the spec parser to recurse with all manner of funny side-effects, I'm not sure that is solvable within the existing spec syntax because of all the historic quirks involved.

Allowing a "sub-package" to share the main package name may get around some while introducing others.
